src/NewFamilyMember.tsx

```diff
@@ -0,0 +1,229 @@
+import { useState } from "react";
+import "./NewFamilyMember.scss";
+
+enum DocumentType {
+  Type = 0,
+  IDCard = 1,
+  FIN = 2,
+  Passport = 3,
+}
+
+const documentValidation: {[key: string]: RegExp} = {
+  [DocumentType.IDCard]: /\w{8}[A-Z]/,
+  [DocumentType.FIN]: /[A-Z]\w{8}[A-Z]/,
+  [DocumentType.Passport]: /[A-Z]{3}\w{6}/,
+}
+
+function NewFamilyMember() {
+  interface FamilyMember {
+    value: undefined | string | DocumentType;
+    set: (value: any) => void | React.Dispatch<React.SetStateAction<any>>;
+    isValid: (value: any) => boolean;
+  }
+  const newFamilyMember: { [key: string]: FamilyMember } = {
+    name: {
+      value: undefined,
+      set: () => {},
+      isValid: (value: string): boolean => {
+        return value.length >= 2 && value.length <= 30 && !!value.match(/^([a-zA-ZÀ-ÿ\u00f1\u00d1]+ )*[a-zA-ZÀ-ÿ\u00f1\u00d1]+$/);
```

Comment:
it might be worth extracting this regex to a variable and give it a meaningful name. What do you think?